dispatch/sqsdispatch.go

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2020 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package dispatch
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/xmidt-org/mimisbrunnr/model"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/semaphore"
+	"github.com/xmidt-org/wrp-go/v2"
+)
+
+type SqsDispatcher struct {
+	Wg        sync.WaitGroup
+	Measures  *Measures
+	Logger    log.Logger
+	Norn      model.Norn
+	Workers   semaphore.Interface
+	SqsClient *sqs.SQS
+}
+
+func NewSqsDispatcher(dc DispatcherConfig, norn model.Norn, transport http.RoundTripper) (D, error) {
+	if dc.QueueSize < defaultMinQueueSize {
+		dc.QueueSize = defaultMinQueueSize
+	}
+
+	if dc.NumWorkers < minMaxWorkers {
+		dc.NumWorkers = minMaxWorkers
+	}
+
+	dispatcher := SqsDispatcher{
+		Norn: norn,
+	}
+	err := validateAWSConfig(norn.Destination.AWSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return dispatcher, nil
+}
+
+func validateAWSConfig(config model.AWSConfig) error {
+	if config.AccessKey == "" {
+		return fmt.Errorf("invalid AWS accesskey")
+	}
+
+	if config.SecretKey == "" {
+		return fmt.Errorf("invalid AWS secretkey")
+	}
+
+	if config.ID == "" {
+		return fmt.Errorf("invalid AWS id")
+	}
+
+	if config.Sqs.QueueURL == "" {
```

Comment:
I think we should validate that the QueueURL is a URI, similar to the check done in the HTTPDispatcher creation.